util/measurement.go

```diff
@@ -0,0 +1,19 @@
+package util
+
+import (
+	"context"
+	"net"
+	"time"
+)
+
+func DelayMeasurement(ctx context.Context, network, address string, timeout time.Duration) (error, time.Duration) {
+	now := time.Now()
+	d := net.Dialer{Timeout: timeout}
+	_, err := d.DialContext(ctx, network, address)
```

Comment:
Since you don't cancel the context here, I think you don't need the context at all and can simply use `d.Dial()` or even `net.DialTimeout`